client/scala/src/main/scala/ai/verta/client/Client.scala

```diff
@@ -115,4 +116,70 @@ class Client(conn: ClientConnection) {
   private def getPersonalWorkspace()(implicit ec: ExecutionContext): String = {
     "personal"
   }
+
+  /** Gets a dataset by its name. If no such dataset exists, creates it.
+   *  @param name name of the dataset.
+   *  @param workspace name of the workspace.
+   *  @return the retrieved or created dataset.
+   */
+  def getOrCreateDataset(name: String, workspace: Option[String] = None)(implicit ec: ExecutionContext): Try[Dataset] =
+    processWorkspace(workspace).flatMap(workspace =>
+      GetOrCreateEntity.getOrCreate[Dataset](
+        get = () => {
+          clientSet.datasetService.DatasetService_getDatasetByName(
+            name = Some(name),
+            workspace_name = Some(workspace)
+          ).flatMap(r => {
+            if (r.dataset_by_user.isDefined)
+              Success(new Dataset(clientSet, r.dataset_by_user.get))
+            else if (r.shared_datasets.isDefined && r.shared_datasets.get.length > 0)
+              Success(new Dataset(clientSet, r.shared_datasets.get(0)))
+            else
+              Failure(new IllegalArgumentException("No dataset with such name exists."))
+          })
+        },
+        create = () => {
+          clientSet.datasetService.DatasetService_createDataset(ModeldbCreateDataset(
+            name = Some(name),
+            workspace_name = Some(workspace)
+          )).map(r => new Dataset(clientSet, r.dataset.get))
+        }
+      )
+    )
+
+  /** Gets a dataset by its id.
+   *  @param id ID of the dataset.
+   *  @return the retrieved dataset, if exists.
+   */
+  def getDatasetById(id: String)(implicit ec: ExecutionContext): Try[Dataset] =
+    clientSet.datasetService.DatasetService_getDatasetById(Some(id))
+      .map(r => new Dataset(clientSet, r.dataset.get))
+
+  /** Gets a dataset by its name.
+   *  @param name name of the dataset.
+   *  @param workspace name of the workspace.
+   *  @return the retrieved dataset, if exists.
+   */
+  def getDatasetByName(name: String, workspace: Option[String]=None)(implicit ec: ExecutionContext): Try[Dataset] =
```

Comment:
Make sure to enable formatting in your IDE
```suggestion
  def getDatasetByName(name: String, workspace: Option[String] = None)(implicit ec: ExecutionContext): Try[Dataset] =
```